lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -40,6 +40,14 @@ def deploy_failed?
       @pods.present? && @pods.all?(&:deploy_failed?)
     end
 
+    def failure_message
+      @pods.map(&:failure_message).compact.uniq.join("\n")
```

Comment:
The `failure_message`s are fairly arbitrary strings, so this could be a bit intensive if there are a ton of failed pods in the set, and in the vast majority of cases, it will only yield one failure message anyway. Wdyt, worth doing over `@pods.find(&:deploy_failed?).failure_message`?